### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by Nyholm**
I noticed that this library does not fully comply with psr7. 

I’ve added integration test to show where it is failing. 

(I’m at the airport atm, will complete this PR later)

### CODE REVIEW

**Comment by Nyholm** [comment]
Oh. It looks like this PR is working as expected. There are just some cs fixes needed

**Comment by Ocramius** [comment]
Picking this up for checking while I'm on travel tomorrow :+1:

File: .travis.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ cache:
 env:
   global:
     - COMPOSER_ARGS="--no-interaction"
-    - COVERAGE_DEPS="php-coveralls/php-coveralls"
+    - DEPENDENCIES=""
```
Comment (weierophinney) [inline .travis.yml]: Technically, we don't need to pre-define globals; the shell will resolve it to `''` if not defined.

File: .travis.yml [hunk 2]
```diff
@@ -62,7 +68,7 @@ install:
   - if [[ $TRAVIS_PHP_VERSION =~ ^5.6 ]]; then travis_retry composer update $COMPOSER_ARGS --with-dependencies $LEGACY_DEPS ; fi
   - if [[ $DEPS == 'latest' ]]; then travis_retry composer update $COMPOSER_ARGS ; fi
   - if [[ $DEPS == 'lowest' ]]; then travis_retry composer update --prefer-lowest --prefer-stable $COMPOSER_ARGS ; fi
-  - if [[ $TEST_COVERAGE == 'true' ]]; then travis_retry composer require --dev $COMPOSER_ARGS $COVERAGE_DEPS ; fi
+  - if ! [ -v "$DEPENDENCIES" ]; then travis_retry composer require --dev $COMPOSER_ARGS $DEPENDENCIES ; fi
```
Comment (weierophinney) [inline .travis.yml]: This should become `if [[ $DEPENDENCIES != '' ]]` to be consistent with our other checks.

Also, please rename it to `INTEGRATION_DEPS` so we know _what kind_ of dependencies we're defining and using here.

File: test/Integration/RequestTest.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+/**
+ * @see       https://github.com/zendframework/zend-diactoros for the canonical source repository
+ * @copyright Copyright (c) 2015-2017 Zend Technologies USA Inc. (http://www.zend.com)
```
Comment (weierophinney) [inline test/Integration/RequestTest.php]: Change this line in all new files to:

```php
 * @copyright Copyright (c) 2018 Zend Technologies USA Inc. (https://www.zend.com)
```

(Copyright extends from when the _file_ is introduced. Also, updates to use SSL-enabled URL.)

**Pull request merged**